### ISSUE TITLE
Getting StackOverflowException at JsonDocument.Parse(content!).RootElement

### ISSUE THREAD

**Issue by usmani93** [IdentityModel/IdentityModel.OidcClient#414]
Hey, I'm using Oidc client in my .Net framework 4.6.2 application to generate token via Okta. Now it is throwing StackOverflowException inside ProtocolResponse at line 67:
```
 // either 200 or 400 - both cases need a JSON response (if present), otherwise error
        try
        {
            if (content!.IsPresent())
            {
                response.Json = JsonDocument.Parse(content!).RootElement; <--- here
            }
        }
```

I'm doing this:
```
            OidcClientOptions oidcOptions = new OidcClientOptions
            {
                //PostLogoutRedirectUri = SsoDefaults.PostLogoutRedirectUri,
                ClientSecret = SsoDefaults.ClientSecret,
                RedirectUri = SsoDefaults.RedirectUri,
                Authority = SsoDefaults.Authority,
                ClientId = SsoDefaults.ClientId,
                Scope = SsoDefaults.Scope,
            };

            OidcClient oidcClient = new OidcClient(oidcOptions);
            AuthorizeState state = await oidcClient.PrepareLoginAsync();
            CallbackManager callbackManager = new CallbackManager(state.State);
            Process.Start(state.StartUrl);
```

And this throws me exception without going into catch, this happens when `AuthorizeState state = await oidcClient.PrepareLoginAsync();` is executed. 

Thanks

**Comment by leastprivilege** [comment]
> **Now** it is throwing StackOverflowException inside ProtocolResponse

Has it ever worked for you? What has changed?

**Comment by usmani93** [comment]
> Has it ever worked for you? What has changed?

Well, When I run the program from visual studio, everything is fine, but when I separate the bin folder, and execute the exe from there it shows me error, sometimes like this exception and some times it says dll not found for Microsoft.Bcl. I have checked the folder contains all these dlls but it is doing this

**Comment by leastprivilege** [comment]
OK - that's a weird dependency problem I would guess. I do not know the answer.

**Comment by usmani93** [comment]
Well, My project structure is like, there is my main project which have class libraries, and one of them was referencing the oidc client project, so I included dependencies for oidc client in all of my projects. Ultimately I removed references from all projects except the one in which oidc was being used, even from app.config and package.config files. And now everything is working good. 
Thankyou for your time.
